### ISSUE TITLE
[Flake Test] testContainerShouldNotMountDirectory

### ISSUE THREAD

**Issue by liangyuanpeng** [etcd-io/jetcd#1232]
```shell
5 tests completed, 2 failed
> Task :jetcd-launcher:test
  io.etcd.jetcd.launcher.test.EtcdClusterStartTest ✘ testContainerShouldNotMountDirectory() (7.2s)

    java.lang.IllegalStateException: Cluster failed to start
        at io.etcd.jetcd.launcher.test.EtcdClusterStartTest.testContainerShouldNotMountDirectory(EtcdClusterStartTest.java:60)
    Caused by: java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image gcr.io/etcd-development/etcd:v3.5.9

    Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image gcr.io/etcd-development/etcd:v3.5.9

    Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception

    Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container

    Caused by: org.testcontainers.shaded.org.awaitility.core.ConditionTimeoutException: Lambda expression in org.testcontainers.containers.GenericContainer: expected the predicate to return <true> but it returned <false> for input of <InspectContainerResponse(args=[--name, etcd0, --advertise-client-urls, http://0.0.0.0:2379, --listen-client-urls, http://0.0.0.0:2379], config=ContainerConfig(attachStderr=false, attachStdin=false, attachStdout=false, cmd=[etcd, --name, etcd0, --advertise-client-urls, http://0.0.0.0:2379, --listen-client-urls, http://0.0.0.0:2379], domainName=, entrypoint=null, env=[ETCD_LOG_LEVEL=debug, PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin, SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt], exposedPorts=[2379/tcp, 2380/tcp], hostName=09e5719c2e72, image=gcr.io/etcd-development/etcd:v3.5.9, labels={org.testcontainers=true, org.testcontainers.lang=java, org.testcontainers.sessionId=ba5695f2-2559-4bf4-9d4a-be7f3286143b, org.testcontainers.version=1.19.0}, macAddress=null, networkDisabled=null, onBuild=null, stdinOpen=false, portSpecs=null, stdInOnce=false, tty=false, user=1001:127, volumes=null, workingDir=/var/lib/etcd, healthCheck=null), created=2023-09-22T17:49:51.995647895Z, driver=overlay2, execDriver=null, hostConfig=HostConfig(binds=[], blkioWeight=0, blkioWeightDevice=null, blkioDeviceReadBps=null, blkioDeviceWriteBps=null, blkioDeviceReadIOps=null, blkioDeviceWriteIOps=null, memorySwappiness=null, nanoCPUs=0, capAdd=null, capDrop=null, containerIDFile=, cpuPeriod=0, cpuRealtimePeriod=0, cpuRealtimeRuntime=0, cpuShares=0, cpuQuota=0, cpusetCpus=, cpusetMems=, devices=null, deviceCgroupRules=null, deviceRequests=null, diskQuota=null, dns=null, dnsOptions=null, dnsSearch=null, extraHosts=[], groupAdd=null, ipcMode=shareable, cgroup=, links=[], logConfig=LogConfig(type=json-file, config={}), lxcConf=null, memory=0, memorySwap=0, memoryReservation=0, kernelMemory=null, networkMode=2fe87db487f98c52c79ab8c20c9080b1b152a831e92c8709f87cb40649a778c2, oomKillDisable=null, init=null, autoRemove=false, oomScoreAdj=0, portBindings={2380/tcp=[Lcom.github.dockerjava.api.model.Ports$Binding;@2d04f21d, 2379/tcp=[Lcom.github.dockerjava.api.model.Ports$Binding;@77c0ae76}, privileged=false, publishAllPorts=false, readonlyRootfs=false, restartPolicy=no, ulimits=null, cpuCount=0, cpuPercent=0, ioMaximumIOps=0, ioMaximumBandwidth=0, volumesFrom=[], mounts=null, pidMode=, isolation=null, securityOpts=null, storageOpt=null, cgroupParent=, volumeDriver=, shmSize=67108864, pidsLimit=null, runtime=runc, tmpFs=null, utSMode=, usernsMode=, sysctls=null, consoleSize=[0, 0], cgroupnsMode=private), hostnamePath=/var/lib/docker/containers/09e5719c2e72e27e0e3490e2f9bc5ee66c13ec9e349ffcc80be83fdd5ad561c6/hostname, hostsPath=/var/lib/docker/containers/09e5719c2e72e27e0e3490e2f9bc5ee66c13ec9e349ffcc80be83fdd5ad561c6/hosts, logPath=/var/lib/docker/containers/09e5719c2e72e27e0e3490e2f9bc5ee66c13ec9e349ffcc80be83fdd5ad561c6/09e5719c2e72e27e0e3490e2f9bc5ee66c13ec9e349ffcc80be83fdd5ad561c6-json.log, id=09e5719c2e72e27e0e3490e2f9bc5ee66c13ec9e349ffcc80be83fdd5ad561c6, sizeRootFs=null, imageId=sha256:3fa6a9[376](https://github.com/etcd-io/jetcd/actions/runs/6277445148/job/17049206195#step:4:377)af360c0b7a9d6ee6c23f4b1af29f57f6faa93cfe4a482dd16898574, mountLabel=, name=/nice_buck, restartCount=0, networkSettings=NetworkSettings(bridge=, sandboxId=e1c732d13cab899968fe71a5b39c42bce79b27dc16f630b9bea01b28d76a2006, hairpinMode=false, linkLocalIPv6Address=, linkLocalIPv6PrefixLen=0, ports={}, sandboxKey=/var/run/docker/netns/e1c732d13cab, secondaryIPAddresses=null, secondaryIPv6Addresses=null, endpointID=, gateway=, portMapping=null, globalIPv6Address=, globalIPv6PrefixLen=0, ipAddress=, ipPrefixLen=0, ipV6Gateway=, macAddress=, networks={95bf82bb-2409-43e5-8642-bd890cbc7512=ContainerNetwork(ipamConfig=null, links=[], aliases=[tc-HWLxQBLn, etcd0, 09e5719c2e72], networkID=2fe87db487f98c52c79ab8c20c9080b1b152a831e92c8709f87cb40649a778c2, endpointId=, gateway=, ipAddress=, ipPrefixLen=0, ipV6Gateway=, globalIPv6Address=, globalIPv6PrefixLen=0, macAddress=)}), path=etcd, processLabel=, resolvConfPath=/var/lib/docker/containers/09e5719c2e72e27e0e3490e2f9bc5ee66c13ec9e349ffcc80be83fdd5ad561c6/resolv.conf, execIds=null, state=InspectContainerResponse.ContainerState(status=exited, running=false, paused=false, restarting=false, oomKilled=false, dead=false, pid=0, exitCode=1, error=, startedAt=2023-09-22T17:49:52.31454672Z, finishedAt=2023-09-22T17:49:52.327452597Z, health=null), volumes=null, volumesRW=null, node=null, mounts=[], graphDriver=GraphDriver(name=overlay2, data=GraphData(rootDir=null, deviceId=null, deviceName=null, deviceSize=null, dir=null)), platform=linux)> within 5 seconds.
```

Failed github action:
- https://github.com/etcd-io/jetcd/actions/runs/6277445148/job/17049206195
- https://github.com/etcd-io/jetcd/actions/runs/6244573932/job/16951700569

**Comment by liangyuanpeng** [comment]
Seems like it's still happen, More investigations will take place.

**Comment by liangyuanpeng** [comment]
Fixed by #1249
